portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -1,26 +1,59 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.temporal.ChronoUnit;
+
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
 @WebServlet("/form-handler")
 public class FormHandlerServlet extends HttpServlet {
```

Comment:
optional: What about a quick doGet() to retrieve all contact info?
```

  @Override
  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
    Query<Entity> query =
        Query.newEntityQueryBuilder().setKind("Contact").setOrderBy(OrderBy.desc("timeStamp")).build();
    QueryResults<Entity> results = datastore.run(query);

    List<Contact> contacts = new ArrayList<>();

    // todo(mseechan): Translate each data store entity to a Java object. 
    // I'll leave this to you, but the ListTasksServlet is a great example 

    Gson gson = new Gson();

    response.setContentType("application/json;");
    response.getWriter().println(gson.toJson(contacts));
  }
```

Reply:
Thanks! I'm currently reworking the example servlet to fit my page. I'll let you know if I run into any issues.